pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -419,7 +419,22 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 		responseHeadersToRemove = append(responseHeadersToRemove, in.RemoveResponseHeaders...)
 		out.ResponseHeadersToRemove = responseHeadersToRemove
 
-		if in.Mirror != nil {
+		if in.Mirror2 != nil {
+			if in.Mirror2.Weight == 0 {
+				in.Mirror2.Weight = 100
```

Comment:
Should we instead disable mirroring if `weight` is set to `0`?

Reply:
yeah, probably, and emit a warning, maybe?

Was trying to keep the current behavior, i.e, user does not supply weight and get 100% routed, but since user is now using another field (`mirror_percent`?) it's expected that they fill all mandatory fields...

Reply:
I'd match it with the behavior of other % based configurations (tcp/http routing, faults, etc.) in Istio. Maybe we can mirror 100% of the traffic if a % is not mentioned i.e. the current behavior you have. Looks like that might be a better choice.